.github/workflows/publish.yml

```diff
@@ -161,25 +163,38 @@ jobs:
           git commit -m "Update CHANGELOG.md for v${{ steps.version.outputs.current-version }} [skip ci]"
           git push origin next
 
-      - name: Sync versions/next.json from `next` to `main`
+      - name: Sync version JSONs from `next-release` to `main`
         if: github.ref_name == 'next-release'
         working-directory: .
         run: |
+          VERSION_FILE="./docs/versions/${{ steps.is-prerelease.outputs.prerelease == 'true' && 'next' || 'latest' }}.json"
           git fetch origin main
           git checkout main
           git pull
-          git checkout origin/next ./docs/versions/next.json
-          git add ./docs/versions/next.json
-          git commit -m "Update versions/next.json for v${{ steps.version.outputs.current-version }}"
+          git checkout origin/next-release $VERSION_FILE
+          git add $VERSION_FILE
+          git commit -m "Update $VERSION_FILE for v${{ steps.version.outputs.current-version }}"
           git push origin main
 
-      # Force push from next to main if it is not a prerelease, and this release is from next-release
+      # Sync next-release to main if it is not a prerelease, and this release is from next-release
       # This happens when eg. next has been tracking 7.1.0-alpha.X, and now we want to release 7.1.0
-      # This will keep release-next, next and main all tracking v7.1.0
-      # - name: Force push ${{ steps.target.outputs.target }} to main
-      #   if: steps.publish-needed.outputs.published == 'false' && steps.target.outputs.target == 'next' && !steps.is-prerelease.outputs.prerelease
-      #   run: |
-      #     git push --force origin ${{ steps.target.outputs.target }}:main
+      # This will keep next-release, next and main all tracking v7.1.0
+      # See "Alternative merge strategies" in https://stackoverflow.com/a/36321787
+      - name: Sync next-release to main
+        if: steps.publish-needed.outputs.published == 'false' && steps.target.outputs.target == 'next' && !steps.is-prerelease.outputs.prerelease
+        working-directory: .
+        run: |
+          git fetch origin next-release
+          git checkout next-release
+          git pull
+          git fetch origin main
+          git checkout main
+          git pull
+          git merge --no-commit -s ours next-release
+          git rm -rf .
+          git checkout next-release -- .
+          git commit -m "Sync next-release to main"
+          git push origin main
```

Comment:
@ndelangen came up with a better approach, because this doesn't get all of `next` history into `main`, it squashes it all into one commit.

Instead we could create a branch from main, called `release-7.1`, and then reset `main` to point to `next` and force push.
This would ensure that the history of `main` is the one from `next` - which is correct - and we'd still be able to find the history for 7.1 by looking at the `release-7.1` branch, because the commits are still intact.